src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/Chunk.java

```diff
@@ -47,13 +50,23 @@ public Chunk(ConnectionImpl connection) {
 			transformer.loadData(chunk, data, bitmask, cache.getAttributesCache().hasSkyLightInCurrentDimension(), full, tiles);
 			ClientBoundPacketData chunkpacket = ClientBoundPacketData.create(PEPacketIDs.CHUNK_DATA);
 			PositionSerializer.writePEChunkCoord(chunkpacket, chunk);
-			ArraySerializer.writeVarIntByteArray(chunkpacket, chunkdata -> {
+			Consumer<ByteBuf> chunkDataProducer = chunkdata -> {
 				transformer.writeLegacyData(chunkdata);
 				chunkdata.writeByte(0); //borders
 				for (TileEntity tile : transformer.remapAndGetTiles()) {
 					ItemStackSerializer.writeTag(chunkdata, true, version, tile.getNBT());
 				}
-			});
+			};
+			if (version.isAfterOrEq(ProtocolVersion.MINECRAFT_PE_1_12)) {
+				ByteBuf tmpBuf = Unpooled.buffer(512);
+				chunkDataProducer.accept(tmpBuf);
+				VarNumberSerializer.writeVarInt(chunkpacket, tmpBuf.readUnsignedByte());
+				chunkpacket.writeByte(0);
+				ArraySerializer.writeVarIntByteArray(chunkpacket, tmpBuf);
```

Comment:
Are you sure this is a byte + byte fixed 0, and not a change to a int length field?

Reply:
I think im going to rewrite this part. I copied what was in steadfast, and it does work, but its stupidly vague as to whats actually going on there. I've been out this whole month, hopefully should have some time the next few weeks to finish up this and get updated with the 1.14 mainline stuff